### PR TITLE
Add some more examples to readme

### PR DESCRIPTION
**Pull request by flatsiedatsie**
- Added basic installation instructions
- Added some more examples to quickly get started.
- Added note about sub_type having different content based on context.
- Changed protocol version in example to 2.2, and enabled persistence option in the example too. It's probably easier to disable than enable.

### CODE REVIEW

**Comment by flatsiedatsie** [comment]
You might also enjoy this:
https://forum.mysensors.org/topic/10342/mozilla-webthings-mysensors-add-on-adapter

File: README.md [hunk 1]
```diff
@@ -7,6 +7,11 @@ Python API for talking to a MySensors gateway (http://www.mysensors.org/). Curre
 - All gateway instances, serial, tcp (ethernet) or mqtt will run in separate threads.
 - As an alternative to running the gateway in its own thread, there are experimental implementations of all gateways using asyncio.
 
+# Installation
+You can easily install it via PIP:
```
Comment (MartinHjelmare) [inline README.md]:
```suggestion
You can easily install it from PyPI:
```

File: README.md [hunk 2]
```diff
@@ -58,16 +64,35 @@ ChildSensor - a child sensor
...
 
 ```python
 s_type = GATEWAY.sensors[23].children[4].type
 values = GATEWAY.sensors[23].children[4].values
 ```
+
+Similarly, printing all the sketch names of the found nodes could look like this:
+
+```
+for nodeIndex in GATEWAY.sensors:
```
Comment (MartinHjelmare) [inline README.md]:
```suggestion
for node in GATEWAY.sensors.values():
    print(node.sketch_name)
```

File: README.md [hunk 3]
```diff
@@ -58,16 +64,35 @@ ChildSensor - a child sensor
...
+    print(str(GATEWAY.sensors[nodeIndex].sketch_name))
```
Comment (MartinHjelmare) [inline README.md]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -58,16 +64,35 @@ ChildSensor - a child sensor
...
+```
+
+Getting a child object inside the event function could be:
+
+```
+try:
+    childObject = GATEWAY.sensors[message.node_id].children[message.child_id]
```
Comment (MartinHjelmare) [inline README.md]:
```suggestion
    if GATEWAY.is_sensor(message.node_id, message.child_id):
        child = GATEWAY.sensors[message.node_id].children[message.child_id]
    else:
        print("Child not available yet.")
```

File: README.md [hunk 5]
```diff
@@ -58,16 +64,35 @@ ChildSensor - a child sensor
...
+
+```
+for nodeIndex in GATEWAY.sensors:
+    print(str(GATEWAY.sensors[nodeIndex].sketch_name))
+```
+
+Getting a child object inside the event function could be:
+
+```
+try:
```
Comment (MartinHjelmare) [inline README.md]:
```suggestion
```

File: README.md [hunk 6]
```diff
@@ -58,16 +64,35 @@ ChildSensor - a child sensor
...
+    childObject = GATEWAY.sensors[message.node_id].children[message.child_id]
+    print(str(childObject))
```
Comment (MartinHjelmare) [inline README.md]:
```suggestion
```

File: README.md [hunk 7]
```diff
@@ -58,16 +64,35 @@ ChildSensor - a child sensor
...
+except Exception as ex:
```
Comment (MartinHjelmare) [inline README.md]:
```suggestion
```

File: README.md [hunk 8]
```diff
@@ -58,16 +64,35 @@ ChildSensor - a child sensor
...
+    print("Child not available yet. Error: " + str(ex))
```
Comment (MartinHjelmare) [inline README.md]:
```suggestion
```

File: README.md [hunk 9]
```diff
@@ -82,7 +107,7 @@ GATEWAY.start_persistence()
 ```
 
 ## Protocol version
-Set the keyword argument `protocol_version` to set which version of the MySensors serial API to use. The default value is `'1.4'`.
+Set the keyword argument `protocol_version` to set which version of the MySensors serial API to use. The default value is `'1.4'` so you probably want to change this to the latest version.
```
Comment (MartinHjelmare) [inline README.md]:
```suggestion
Set the keyword argument `protocol_version` to set which version of the MySensors serial API to use. The default value is `'1.4'`. Set the `protocol_version` to the version you're using.
```

**Comment by flatsiedatsie** [comment]
Is there anything left to do? Anything I should do?

**Comment by MartinHjelmare** [comment]
No I approved before the build was done. I don't like to merge until bild is done, even though this is just an update of the readme.

Thanks for pinging me again.

**Pull request merged**